src/Expander/Primitives.hs

```diff
@@ -512,90 +509,41 @@ typeCase t dest stx = do
 type TypePrim =
   (Kind -> SplitTypePtr -> Syntax -> Expand (), TypePatternPtr -> Syntax -> Expand ())
 
-baseType :: (forall a . TyF a) -> TypePrim
-baseType ctor = (implT, implP)
-  where
-    implT k dest stx = do
-      equateKinds stx KStar k
-      _actualName <- mustBeIdent stx
```

Comment:
We are no longer checking that the `stx` must be an identifier; on the contrary, we are now checking that it's a singleton list.